lib/broccoli/ember-app.js

```diff
@@ -41,6 +41,15 @@ var isNull        = require('ember-cli-lodash-subset').isNull;
 var Funnel        = require('broccoli-funnel');
 var funnelReducer = require('broccoli-funnel-reducer');
 
+function mergeTrees(_trees, options) {
```

Comment:
@nathanhammond i started with that, but it confuses the internal node wrappers, and annotations for wired up incorrectly.  I do think eventually it should be merely a broccoli primitive (maybe in BMT or maybe???), which elides pointless things.